crates/bevy_reflect/src/serde/de/mod.rs

```diff
@@ -514,6 +517,197 @@ mod tests {
+                _: &TypeRegistry,
+                deserializer: D,
+            ) -> Result<Result<Box<dyn PartialReflect>, D>, D::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                if registration.type_id() == TypeId::of::<i32>() {
+                    let _ = deserializer.deserialize_ignored_any(IgnoredAny);
+                    Ok(Ok(Box::new(123_i32)))
+                } else if registration.type_id() == TypeId::of::<i64>() {
+                    let _ = deserializer.deserialize_ignored_any(IgnoredAny);
+                    Ok(Ok(Box::new(456_i64)))
+                } else {
+                    Ok(Err(deserializer))
+                }
+            }
+        }
+
+        let expected = Foo { bar: 123, qux: 456 };
+
+        let input = r#"(
+            bar: 0,
+            qux: 0,
+        )"#;
+
+        let mut registry = get_registry();
+        registry.register::<Foo>();
+        let registration = registry.get(TypeId::of::<Foo>()).unwrap();
+        let mut processor = FooProcessor;
+        let reflect_deserializer =
+            TypedReflectDeserializer::with_processor(registration, &registry, &mut processor);
+        let mut ron_deserializer = ron::de::Deserializer::from_str(input).unwrap();
+        let dynamic_output = reflect_deserializer
+            .deserialize(&mut ron_deserializer)
+            .unwrap();
+
+        let output =
+            <Foo as FromReflect>::from_reflect(dynamic_output.as_ref().as_partial_reflect())
+                .unwrap();
+        assert_eq!(expected, output);
+    }
+
+    #[test]
+    fn should_propagate_processor_deserialize_error() {
+        struct ErroringProcessor;
+
+        impl ReflectDeserializerProcessor for ErroringProcessor {
+            fn try_deserialize<'de, D>(
+                &mut self,
+                registration: &TypeRegistration,
+                _: &TypeRegistry,
+                deserializer: D,
+            ) -> Result<Result<Box<dyn PartialReflect>, D>, D::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                if registration.type_id() == TypeId::of::<i32>() {
+                    Err(serde::de::Error::custom("my custom deserialize error"))
+                } else {
+                    Ok(Err(deserializer))
+                }
+            }
+        }
+
+        let registry = get_registry();
+
+        let input = r#"{"i32":123}"#;
+        let mut deserializer = ron::de::Deserializer::from_str(input).unwrap();
+        let mut processor = ErroringProcessor;
+        let reflect_deserializer = ReflectDeserializer::with_processor(&registry, &mut processor);
+        let error = reflect_deserializer
+            .deserialize(&mut deserializer)
+            .unwrap_err();
+
+        #[cfg(feature = "debug_stack")]
+        assert_eq!(
+            error,
+            ron::Error::Message("my custom deserialize error (stack: `i32`)".to_string())
+        );
```

Comment:
How does the processor affect the `debug_stack` for deeper nestings?

For example:

```
Foo -> Bar (w/ Processor) -> Baz -> i32
```

Does it print the full stack:

```
Foo -> Bar -> Baz -> i32
```

Or does it skip the processed type:

```
Foo -> Bar -> i32
```

Or something else?

Reply:
Because the processor takes ownership of the deserializer here, the processor would have to manually create a `ReflectDeserializer::new`, resetting the type stack. If the processor overrides deserialization of `Bar`, it would look like `Baz -> i32`.
Do we need some smarter mechanism for handling the type stack here?

Reply:
This also made me realise that if you have a processor (or a `DeserializeWithRegistry`) impl that creates a new `ReflectDeserializer`, it won't have the type stack context or the processor context with it. So in the example where we perform extra logic when loading `Handle`s, if one of the reflected types implements `DeserializeWithRegistry` and creates its own `TypedReflectDeserializer`, that deserializer won't have the processor, and any handles under that won't be properly loaded.